File: src/dto/PrintBlockTextRaw.ts
import type { PrintBlockText } from './PrintBlock';

export interface PrintBlockTextRaw extends PrintBlockText {
  type: 'textRaw';
  /** @default GBK */
  encoding: string;
  /** @default 0 */
  codepage: number;
  /** @default 0 */
  scaleWidth: number;
  /** @default 0 */
  scaleHeight: number;
  /** @default 0 */
  fontStyle: number;
  /**
   * @enum { Left = 0, Center = 1, Right = 2 }
   * @default 0
   */
  align: number;
}

File: android/src/main/java/blue/muffin/btprinter/zj/dto/PrintBlockTextRaw.java
package blue.muffin.btprinter.zj.dto;

import org.json.JSONObject;

public final class PrintBlockTextRaw extends PrintBlockText {

    private final String encoding;
    private final int codepage;
    private final int scaleWidth;
    private final int scaleHeight;
    private final int fontStyle;
    private final int align;

    public PrintBlockTextRaw(JSONObject block, PrintBlockType type) {
        super(block, type);
        this.encoding = block.optString("encoding", "GBK");
        this.codepage = block.optInt("codepage", 0);
        this.scaleWidth = block.optInt("scaleWidth", 0);
        this.scaleHeight = block.optInt("scaleHeight", 0);
        this.fontStyle = block.optInt("fontStyle", 0);
        this.align = block.optInt("align", 0);
    }

    public boolean isInvalid() {
        return value == null || value.isEmpty();
    }

    public String getEncoding() {
        return encoding;
    }

    public int getCodepage() {
        return codepage;
    }

    public int getScaleWidth() {
        return scaleWidth;
    }

    public int getScaleHeight() {
        return scaleHeight;
    }

    public int getFontStyle() {
        return fontStyle;
    }

    public int getAlign() {
        return align;
    }
}
